scene/gui/code_edit.cpp

```diff
@@ -914,6 +914,9 @@ void CodeEdit::do_unindent() {
 		int cl = get_caret_line(c);
 		const String &line = get_line(cl);
 
+		if (cc < 0 || (cc - 1) >= line.size())
```

Comment:
We use brackets even for single line if blocks, to have a consistent style.

Reply:
`cc < 0` is also already handled above. So no need to do it here.
In general I'm wondering if this is the right location to handle this. 
As far as I can tell, `add_caret` is already clamping the value. So a bit confusing here...

Reply:
You're right about the `< 0` necessity, but I think we should definitely check before string indexing an invalid indice since it at least ensures us some graceful failure. 

I will also look for other cases where we might need to ensure proper clamping.